db/migrate/20190409164425_validate_not_null_constriants.rb

```diff
@@ -0,0 +1,12 @@
+class ValidateNotNullConstriants < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+  def up
+    execute "ALTER TABLE assignments VALIDATE CONSTRAINT assignments_force_check;"
+    execute "ALTER TABLE previous_assignments VALIDATE CONSTRAINT previous_assignments_force_check;"
+    execute "ALTER TABLE bulk_assignments VALIDATE CONSTRAINT bulk_assignments_force_check;"
```

Comment:
Here's the validate, which takes a long time, but also only out a very ungrabby lock - only conflicts with DDL on the table.